src/hashes.rs

```diff
@@ -1,110 +1,339 @@
-/// List of types currently supported in the multihash spec.
-///
-/// Not all hash types are supported by this library.
-#[derive(PartialEq, Eq, Clone, Debug, Copy, Hash)]
-pub enum Hash {
+use blake2b_simd::Params as Blake2b;
+use blake2s_simd::Params as Blake2s;
+use digest::Digest;
+use sha1::Sha1 as Sha1Hasher;
+use sha2::{Sha256, Sha512};
+use tiny_keccak::{Hasher, Keccak, Sha3};
+
+use crate::digests::{wrap, DynMultihashDigest, Multihash, MultihashDigest};
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Code {
+    /// Make it possible to use a custom code that is not part of the enum yet
+    Custom(u64),
+    /// Hashes that are immplemented in this crate
+    Implemented(ImplementedCode),
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ImplementedCode {
```

Comment:
Thoughts on removing this type and just moving the implemented codes into the Code enum? Seems redundant and annoying to use a seperate type for this. Something like https://github.com/austinabell/rust-multihash/commit/da0d002d5ec3c3e5e64105ddee175f5d2839284c perhaps?

Reply:
Good point and good idea. I wanted that, but you can't mix `Custom(u64)` together with definitions like `Sha2_512 = 0x13`. But what I haven't thought of was doing it your way. :)